dts/bindings/retention/zephyr,retention-driver-state.yaml

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) 2024 SILA Embedded Solutions GmbH
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Driver state retention, which has a retention area as a parent. This is an
+  example containing two drivers. The first one stores one byte, the second one
+  eight. It is strongly recommended to use a prefix and a checksum for the
+  parent retention area to ensure that only valid settings are retained for the
+  drivers.
+
+  retention0: retention@0 {
+    compatible = "zephyr,retention";
+    status = "okay";
+    reg = <0x0 0x40>;
+    prefix = [04 fa];
+    checksum = <1>;
```

Comment:
This would not work with the state code, the state code reads and writes the state from 0 and ignores the reg address, this means if you have one retention area and 4 driver state retention parts, at offsets 0x0, 0x1, 0x2 and 0x3 (ignoring checksums and prefixes), then all of them will actually read and write to 0x0. If you wanted to supported partitioning in the driver state code then you would need to use the reg address and pass that to the retention read/write function